generate.py

```diff
@@ -724,3 +733,185 @@ def normalize(code):
 ---
 { content }
 ''')
+
+
+### Generate Event files
+for event in EVENTS:
+
+  name = event['name']
+
+  # Start and end dates are necessary for listing the events on events.md
+  if not isinstance(event['startDate'], str):
+    raise Exception(event['startDate'])
+
+  if not isinstance(event['endDate'], str):
+    raise Exception(event['endDate'])
+
+  if event.get('blockquotes', None):
+    if not isinstance(event['blockquotes'], list):
+      raise Exception(event['blockquotes'])
+
+  if event.get('urlsAfterBlockquote', None):
+    if not isinstance(event['urlsAfterBlockquote'], list):
+      raise Exception(event['urlsAfterBlockquote'])
+
+  if event.get('linkUrl', None):
+    if not isinstance(event['linkUrl'], list):
+      raise Exception(event['linkUrl'])
+
+  if event.get('impDatesHeaderH4', None):
+    if not isinstance(event['impDatesHeaderH4'], list):
+      raise Exception(event['impDatesHeaderH4'])
+
+  if event.get('names', None):
+    if not isinstance(event['names'], list):
+        raise Exception(event['names'])
+
+  if event.get('speakHeadcontent', None):
+    if not isinstance(event['speakHeadcontent'], list):
+        raise Exception(event['speakHeadcontent'])
+
+  if event.get('wsTitles', None):
+    if not isinstance(event['wsTitles'], list):
+        raise Exception(event['wsTitles'])
+
+  if event.get('sharedTasksUrls', None):
+    if not isinstance(event['sharedTasksUrls'], list):
+      raise Exception(event['sharedTasksUrls'])
+
+  if event.get('scheduleDays', None):
+    if not isinstance(event['scheduleDays'], list):
+      raise Exception(event['scheduleDays'])
+
+  if event.get('cfpContent', None):
+    if not isinstance(event['cfpContent'], list):
+      raise Exception(event['cfpContent'])
+
+  frontmatter = {
+    'autogenerated': True,
+    'autogenerated_from': 'events.json',
+    'parent': 'Events',
+    'nav_order': int(event['startDate'].split('-')[0]),
+    'layout': 'event',
+    **event
+  }
+
+  # Doesn't create a file if it's an external event, as external events should not have an id
+  if not event.get('id', None):
+    continue
+
+  slug = slugify(name)
+  filepath = f'events/{ slug }.md'
+
+  content = read_content(filepath)
+
+  with open(filepath, 'w', encoding='utf-8') as f:
+    f.write(f'''\
+---
+{ yaml.dump(frontmatter, sort_keys=False) }
+---
+{content}
+''')
+
+
+### Generate WMT event files
+for event in WMT_EVENTS:
+
+  name = event['name']
+
+  # Start and end dates are necessary for listing the events on events.md
+  if not isinstance(event['startDate'], str):
+    raise Exception(event['startDate'])
+
+  if not isinstance(event['endDate'], str):
```

Comment:
Maybe if endDate is missing, we assume it's just a single day event?

Reply:
Yes we can